src/engine/source/builder/builders/opBuilderKVDB.cpp

```diff
@@ -0,0 +1,181 @@
+        key = json::formatJsonPath(key.substr(1));
+        isReference = true;
+    }
+
+    // Return Lifter
+    return [target, &kvdb, key, isReference](types::Observable o)
+    {
+        // Append rxcpp operation
+        return o.map(
+            [target, &kvdb, key, isReference](types::Event e)
+            {
+                // Get DB key
+                std::string dbKey;
+                if (isReference){
+                    try
+                    {
+                        auto value = &e->get(key);
+                        if (value && value->IsString()){
+                            dbKey = value->GetString();
+                        }
+                        else {
+                            return e;
+                        }
+                    }
+                    catch (std::exception & ex)
+                    {
+                        // TODO Check exception type
+                        return e;
+                    }
+                }
+                else {
+                    dbKey = std::move(key);
+                }
+
+                // Get value from the DB
+                std::string dbValue = kvdb.read(dbKey);
+                if (dbValue.empty()) {
+                    return e;
+                }
+
+                // Create and add string to event
+                try
+                {
+                    e->set(target,
+                        rapidjson::Value(dbValue.c_str(), e->m_doc.GetAllocator()).Move());
+                }
+                catch (std::exception & ex)
+                {
+                    // TODO Check exception type
+                    return e;
+                }
+
+                return e;
+            });
+    };
+}
+
+types::Lifter opBuilderKVDBExistanceCheck(const types::DocumentValue & def, bool checkExist)
+{
+    // Get key of the match
+    std::string key {json::formatJsonPath(def.MemberBegin()->name.GetString())};
+
+    // Get the raw value of parameter
+    if (!def.MemberBegin()->value.IsString())
+    {
+        throw std::runtime_error(
+            "Invalid parameter type for kvdb matching operator (str expected)");
+    }
+
+    // Parse parameters
+    std::string params {def.MemberBegin()->value.GetString()};
+    auto parametersArr {utils::string::split(params, '/')};
+    if (parametersArr.size() != 2)
+    {
+        throw std::runtime_error("Invalid number of parameters for kvdb matching operator");
+    }
+
+    KVDBManager& kvdbManager = KVDBManager::get();
+    KVDB& kvdb = kvdbManager.getDB(parametersArr[1]);
+    if (kvdb.getState() != KVDB::State::Open)
+    {
+        auto msg = fmt::format("[{}] DB isn't available for usage", parametersArr[1]);
+        throw std::runtime_error(std::move(msg));
+    }
+
+    // Return Lifter
+    return [&kvdb, key, checkExist](types::Observable o)
+    {
+        // Append rxcpp operations
+        return o.filter(
+            [&kvdb, key, checkExist](types::Event e)
```

Comment:
in the future some one could delete the db somewhere else (or update it) so this reference could stay dangling

Reply:
It's true. We will implement a different approach in https://github.com/wazuh/wazuh/issues/12935 where DB can´t be deleted until every reference is deleted.